hpx/runtime/naming/name.hpp

```diff
@@ -818,6 +826,8 @@ namespace hpx { namespace naming
 
             boost::detail::atomic_count count_;
             id_type_management type_;
+
+            mutable std::list<naming::gid_type> new_gids_;
```

Comment:
I think this adds overheads (in terms of memory requirements) which is not acceptable. There will be probably 1% of all ids which require special handling during serialization, but all of them will have a `std::list<gid_type>` member. Also, this does not make sense conceptually, id_type_impl is derived from gid_type and also holds a list of those? That smells.